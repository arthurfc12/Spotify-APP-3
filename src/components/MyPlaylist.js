import axios from 'axios';
import React, { Component } from 'react';
import './App.css';

class MyPlaylist extends Component {
	constructor(props) {
		super(props);
		this.state = {
			playlists: ''
		};
		this.handlePost = this.handlePost.bind(this);
	}

	componentWillMount() {
		const requestOptions = {
			headers: {
				Authorization: 'Bearer ' + this.props.token
			}
		};
		axios.get('https://api.spotify.com/v1/me/playlists', requestOptions).then((response) => {
			this.setState({ playlists: response.data.items });
			console.log(response.data.items);
		});
	}

	handlePost(id) {
		console.log(this.props.token);

		const data = {
			uris: [ this.props.musicUri ]
		};

		const headers = {
			Authorization: 'Bearer ' + this.props.token,
			Accept: 'application/json'
		};

		// console.log('aquiii', `https://api.spotify.com/v1/playlists/${id}/tracks`);
		// console.log('AQUIIIIII', `https://api.spotify.com/v1/playlists/${id}/tracks`, data, { headers: headers });

		axios
			.post(`https://api.spotify.com/v1/playlists/${id}/tracks`, data, {
				headers: headers
			})
			.then((response) => {
				// this.setState({ playlists: response.data.items });
				console.log(response.data);
				alert('Música adicionada com sucesso!');
				window.location.reload(false);
			})
			.catch((error) => {
				console.log(error.response.data);
				alert('Você não pode adicionar em uma playlist que não é sua :(');
			});
	}
	render() {
		const playlists = this.state.playlists;
		var post = this.handlePost;
		return (
			<div>
				<h4>Escolha sua playlist</h4>

				{this.state.playlists !== '' ? (
					playlists.map((playlist, idx) => {
						return (
							<div key={idx}>
								<button
									onClick={() => {
										post(playlist.id);
									}}
									className="gameOpt"
									key={idx}
								>
									<div className="btntext">{playlist.name}</div>
								</button>
							</div>
						);
					})
				) : (
					<div> Carregando</div>
				)}
			</div>
		);
	}
}

export default MyPlaylist;
