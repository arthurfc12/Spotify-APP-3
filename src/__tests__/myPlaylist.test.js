import React from 'react';
import { render, wait, cleanup, waitFor } from '@testing-library/react';
import mockedAxios from 'axios';
import MyPlaylist from '../components/MyPlaylist';
import { configure, shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

afterEach(cleanup);

it('should render the playlists names on the buttons', async () => {
	const data = {
		data: {
			items: [
				{
					name: 'Pop songs',
					id: 1
				},
				{
					name: 'Rock songs',
					id: 2
				}
			]
		}
	};

	mockedAxios.get.mockResolvedValueOnce(data);
	const app = shallow(<MyPlaylist />);
	// expect(app.find(App)).toHaveLength(1);

	await waitFor(() => {
		// expect(mockedAxios.get).toHaveBeenCalledTimes(1);
		expect(app.find('button').at(0).text()).toEqual('Pop songs');
		expect(app.find('button').at(1).text()).toEqual('Rock songs');
	});
});

it('should render the playlists ', async () => {
	const data = {
		data: {
			items: [
				{
					name: 'Pop songs',
					id: 1
				},
				{
					name: 'Rock songs',
					id: 2
				}
			]
		}
	};

	mockedAxios.get.mockResolvedValueOnce(data);
	const app = mount(<MyPlaylist />);
	// expect(app.find(App)).toHaveLength(1);

	await waitFor(() => {
		expect(toJson(app)).toMatchSnapshot();
	});
});
