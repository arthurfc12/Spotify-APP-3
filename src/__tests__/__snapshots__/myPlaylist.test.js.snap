// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the playlists  1`] = `
<MyPlaylist>
  <div>
    <h4>
      Escolha sua playlist
    </h4>
    <div>
       Carregando
    </div>
  </div>
</MyPlaylist>
`;
